### ISSUE TITLE
Add Currency support (MerchantAccount)

### ISSUE THREAD

**Issue by kwestground** [nopSolutions/braintree-plugin-for-nopcommerce#1]
For currency support Braintree requires MerchantAccountId in Transaction.
My suggestion is to add an Generic Attribute to Currency for storing MerchantAccountId.

**BrainTreePaymentProcessor.cs**
```csharp
    public ProcessPaymentResult ProcessPayment(ProcessPaymentRequest processPaymentRequest)
    {
        // ...

        //new transaction request
        var transactionRequest = new TransactionRequest
        {
            Amount = processPaymentRequest.OrderTotal
        };

        // New code below:
        // get order
        var order = _orderService.GetOrderByGuid(processPaymentRequest.OrderGuid);

        // get currency
        var currency = _currencyService.GetCurrencyByCode(order.CustomerCurrencyCode);
        var currencyMerchantId = currency.GetAttribute<string>(CurrencyMerchantAccountId);
        if (!string.IsNullOrEmpty(currencyMerchantId))
        {
            transactionRequest.MerchantAccountId = currencyMerchantId;
        }

        // ...
    }
```
https://articles.braintreepayments.com/get-started/currencies#multi-currency-setups
https://developers.braintreepayments.com/guides/transactions/dotnet#specify-merchant-account

**Comment by IvayloIliev** [comment]
I am interested in this, too.

**Comment by skoshelev** [comment]
Closed #1
